docs/proposals/job_controller.md

```diff
@@ -0,0 +1,226 @@
+* "execTimeout": (not restricted by default) maximal duration a container within a pod started by the Job controller is allowed to run. Reaching this limit leads to the container to be killed.
+* "PodCount": (default to 1) number of pods started by a single run of the Job.
+* "allowConcurrentRun": (default to true) dictates if concurrent runs of this Job are allowed or not.
+
+* "schedule": (optional) sub structure defining scheduling details 
+  * "timeSpec": describes the job schedule, expressed (using the ISO 8601) with a 3 parts dash separated string: "R[n]/[start date]/[duration]"
+  The first part gives the number of repetitions of the job (omitting this number means infinite repetitions)
+  The second part gives the starting time of the scheduling period.
+  The third part is the duration between successive runs of the scheduled pod(s).
+  * "suspendOnFailure": dictated if job schedule should be suspended in case of failure of the last completed job run.
+  * "skipOutdated": dictates if pending scheduled job runs should be discarded in favour of the latest one. This covers the case of a job having suspended its execution after a failed run, with several schedule points reached in the meantime (or the case of a long lasting run with concurrent execution disable). Instead of stacking pending executions, only the latest one is kept (and intermediate pending runs discarded).
+
+**Implementation details**
+
+Job objects are stored in a dedicated registry directory key (separated from the directory key used for replication controllers).
+
+Similarly to replication controllers, a dedicated manager takes care of all jobs existing in the system, performing the following tasks:
+
+* Executes a loop on the full list of defined jobs, calling on each a defined synchronization call-back, each execution loop being spaced by a configuration defined time interval (with default value of 5 seconds).
+* Performs a (recursive) watch of the registry directory key containing jobs definition. For each job newly defined or updated in the system, the same synchronization routine is invoked.
+
+This job manager provides a way for each job defined with schedule information to setup a dedicated timer for next schedule time. This allows a better accuracy of start execution time for pods launched by jobs and remove dependency on the interval value set for the synchronization loop.
+
+The maximal restart count and maximal execution time for pods containers started by a job will be enforced at kubelet level, using dedicated resource limits added by the job in pod's containers definition. Two new resource types ("execution time" and "execution count") are introduced, that are used to add in the container's resources limits the figures defined in job specification and taken into account by the kubelet module (container is killed if the execution time is reached and restarted in case of failure exit up to this maximal count).
```

Comment:
It's not clear to me if the `execution count` is applied to all Jobs generated from single JobController, or rather to a single Job instance? I assume the latter...